sky/shell/tracing_controller.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef __SKY_SHELL_TRACING_CONTROLLER__
+#define __SKY_SHELL_TRACING_CONTROLLER__
+
+#include "base/files/file.h"
+#include "base/macros.h"
+#include "base/memory/ref_counted_memory.h"
+#include "mojo/common/data_pipe_drainer.h"
+#include "mojo/public/cpp/system/data_pipe.h"
+#include "sky/shell/shell_view.h"
+
+#include <memory>
+
+namespace sky {
+namespace shell {
+
+class TracingController : public mojo::common::DataPipeDrainer::Client {
+ public:
+  TracingController();
+  ~TracingController() override;
+
+  void RegisterShellView(ShellView* view);
+  void UnregisterShellView(ShellView* view);
+
+  // Enable tracing in base as well as the dart isolates attached to the shell
+  // views
+  void StartTracing();
+
+  // Stop tracing in base as well as the dart isolates attached to shell views
+  // and dump the resulting trace to the specified path. Traces from various
+  // sources are separated by a NULL character in the resulting file and must
+  // be merged before viewing in the trace viewer
+  void StopTracing(const base::FilePath& path);
+
+ private:
+  std::unique_ptr<mojo::common::DataPipeDrainer> drainer_;
+  std::unique_ptr<base::File> trace_file_;
+  ShellView* view_;
```

Comment:
I'd add a TODO about making this a list.  Currently it's kind of random which one will be stored here.